packages/mux-player/src/index.ts

```diff
@@ -81,6 +76,68 @@ class MuxPlayerInternal {
     this._deinitResizing();
   }
 
+  get _fragments() {
+    return this._template.fragments;
+  }
+
+  _renderChrome() {
+    if (this._playerSize != getPlayerSize(this.el)) {
+      this._playerSize = getPlayerSize(this.el);
+      this._fragments.chromeRenderer.render(
+        getProps(this.el, {
+          ...this._state,
+          isDialogOpen: (
+            this._fragments.dialogContent.parentNode as MxpDialog
+          ).hasAttribute("open"),
+        })
+      );
+    }
+  }
+
+  _initResizing() {
+    this._resizeObserver = new ResizeObserver(() => this._renderChrome());
+    this._resizeObserver.observe(this.el);
+  }
+
+  _deinitResizing() {
+    this._resizeObserver?.disconnect();
+  }
+
+  _setUpErrors(el: MuxPlayerElement) {
+    if (el.video?.hls) {
+      const Hls: any = el.video.hls.constructor;
+      el.video.hls.on(Hls.Events.ERROR, (_event: any, data: any) => {
+        if (data.fatal) {
+          switch (data.type) {
+            case Hls.ErrorTypes.NETWORK_ERROR:
+              this._state.dialog = {
+                title: "Network Error",
+                message:
+                  "A network error occurred. Please reload the player and try again.",
+              };
+              break;
+            case Hls.ErrorTypes.MEDIA_ERROR:
+              this._state.dialog = {
+                title: "Media Error",
+                message:
+                  "A media error occurred. Please reload the player and try again.",
+              };
+              break;
+            default:
+              this._state.dialog = {
+                title: "Error",
+                message:
+                  "An error occurred. Please reload the player and try again.",
```

Comment:
this is fantastic 💯

Reply:
Questions about the mpx-dialog error state:

1. Can we have an easy way (& document this) for developers using Mux Player to simulate the error state. As a developer adding Mux Player in my application I want to be able to visually see what the error state looks like.
2. I don't recall seeing a design for an error state, or any kind of dialog. Is that something we need to loop in some design help on?

Reply:
1. good call, I'll see if I can add this
2. yes, true I kept it really basic. would be very helpful if we had a design for this.